addons/mail/static/src/new/thread/thread_service.js

```diff
@@ -601,6 +608,68 @@ export class ThreadService {
         }
         return message;
     }
+
+    /**
+     * @param {Thread} thread
+     */
+    localMessageUnreadCounter(thread) {
+        // By default trust the server up to the last message it used
+        // because it's not possible to do better.
+        let baseCounter = thread.serverMessageUnreadCounter;
+        let countFromId = thread.serverLastSeenMsgBySelf ? thread.serverLastSeenMsgBySelf : 0;
+        const orderedMessages = this.message
+            .sortMessages(thread)
+            .map((messageId) => this.store.messages[messageId]);
+        // But if the client knows the last seen message that the server
+        // returned (and by assumption all the messages that come after),
+        // the counter can be computed fully locally, ignoring potentially
+        // obsolete values from the server.
+        const lastSeenByCurrentPartnerMessageId = this.lastSeenByCurrentPartnerMessageId(
```

Comment:
Replace `CurrentPartner` to `Self`. This reduces noise a bit, which improves readability.